src/WebJobs.Script/Workers/Rpc/FunctionRegistration/RpcFunctionInvocationDispatcher.cs

```diff
@@ -103,10 +103,22 @@ internal class RpcFunctionInvocationDispatcher : IFunctionInvocationDispatcher
 
         internal ConcurrentStack<WorkerErrorEvent> LanguageWorkerErrors => _languageWorkerErrors;
 
-        internal int MaxProcessCount => _maxProcessCount;
-
         internal IWebHostRpcWorkerChannelManager WebHostLanguageWorkerChannelManager => _webHostLanguageWorkerChannelManager;
 
+        private async Task<int> GetMaxProcessCount()
+        {
+            if (_workerConcurrencyOptions != null && !string.IsNullOrEmpty(_workerRuntime))
+            {
+                var workerConfig = _workerConfigs.Where(c => c.Description.Language.Equals(_workerRuntime, StringComparison.InvariantCultureIgnoreCase)).FirstOrDefault();
+                if (workerConfig != null)
+                {
+                    return _environment.IsWorkerDynamicConcurrencyEnabled() ? _workerConcurrencyOptions.Value.MaxWorkerCount : workerConfig.CountOptions.ProcessCount;
+                }
+            }
+
+            return (await GetAllWorkerChannelsAsync()).ToArray().Count();
```

Comment:
We should be able to skip `ToArray` here

Reply:
Yup, you're right - I don't know why I had to do that in the first place xD